v1/translate.yaml

```diff
@@ -0,0 +1,254 @@
+info:
+  version: '1.0.0'
+  title: Content Translation API
+  description: Content Translation API
+  termsofservice: https://github.com/wikimedia/restbase
+  license:
+    name: Apache licence, v2
+    url: https://www.apache.org/licenses/LICENSE-2.0
+paths:
+  /machine/{from}/{to}{/provider}:
+    post:
+      tags: ['Translate']
+      summary: Machine-translate content
+      description: |
+        Fetches the machine translation for the posted content from the source
+        to the destination language.
+
+        Stability: [unstable](https://www.mediawiki.org/wiki/API_versioning#Unstable)
+      consumes:
+        - application/x-www-form-urlencoded
+      produces:
+        - application/json
+      parameters:
+        - name: from
+          in: path
+          description: The source language code
+          type: string
+          required: true
+        - name: to
+          in: path
+          description: The target language code
+          type: string
+          required: true
+        - name: provider
+          in: path
+          description: The machine translation provider id
+          type: string
+          required: false
+          enum:
+           - Apertium
```

Comment:
I am not opposed to allow case insensitive matching, and I don't think it causes caching issues here. I'm just saying that my expected behavior is that clients should either fetch the languages from our API and return the same keys byte for byte, or the users defining them in the configuration should also check from our API and copy byte for byte.